cmd/create.go

```diff
@@ -0,0 +1,254 @@
+			return
+		}
+
+		// Create a public ip
+		fmt.Println("create public ip resource")
+		fmt.Println("for EdgeGateway name: " + gwName)
+
+		job, err := c.V1.PublicIP.New(gwName)
+		if err != nil {
+			fmt.Println("Unable to create public ip", err)
+			return
+		}
+		err = job.Wait(5, 300)
+		if err != nil {
+			fmt.Println("Error during public ip creation !!", err)
+			return
+		}
+		fmt.Println("public ip resource created successfully !")
+	},
+}
+
+// createCmd represents the create command
+var createVDCCmd = &cobra.Command{
+	Use:     "vdc",
+	Short:   "Create an vdc",
+	Example: "vdc create --name <vdc name>",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		// Check if time flag is set
+		if cmd.Flag("time").Value.String() == "true" {
+			defer timeTrack(time.Now(), cmd.CommandPath())
+		}
+
+		// Get the vdc name from the command line
+		vdcName, err := cmd.Flags().GetString("name")
+		if err != nil {
+			fmt.Println("Malformed VDC name", err)
+			return
+		}
+
+		// Create the vdc
+		fmt.Println("create vdc resource (with basic value)")
+		fmt.Println("vdc name: " + vdcName)
+
+		_, err = c.V1.VDC().New(&infrapi.CAVVirtualDataCenter{VDC: infrapi.CAVVirtualDataCenterVDC{
+			Name:                vdcName,
+			ServiceClass:        "STD",
+			BillingModel:        "PAYG",
+			CPUAllocated:        22000,
+			VCPUInMhz:           2200,
+			Description:         "vdc created by cloudavenue-cli",
+			MemoryAllocated:     30,
+			DisponibilityClass:  "ONE-ROOM",
+			StorageBillingModel: "PAYG",
+			StorageProfiles: []infrapi.StorageProfile{
+				{
+					Class:   "gold",
+					Limit:   500,
+					Default: true,
+				},
+			},
+		},
+		})
+
+		if err != nil {
+			fmt.Println("Error from vdc", err)
+			return
+		}
+
+		fmt.Println("vdc resource created successfully !")
+	},
+}
+
+// createCmd represents the create command
+var createEdgeGatewayCmd = &cobra.Command{
+	Use:     "edgegateway",
+	Short:   "Create an edgeGateway",
+	Aliases: []string{"gw", "egw"},
+	Example: "edgegateway create --vdc <vdc name> [--t0 <t0 name>]",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		// Check if time flag is set
+		if cmd.Flag("time").Value.String() == "true" {
+			defer timeTrack(time.Now(), cmd.CommandPath())
+		}
+
+		// Get the vdc name from the command line
+		vdc, err := cmd.Flags().GetString("vdc")
+		if err != nil {
+			fmt.Println("Error from VDC", err)
+			return
+		}
+
+		// Get the t0 name
+		t0s, err := c.V1.T0.GetT0s()
+		if err != nil {
+			fmt.Println("Error from T0 List", err)
+			return
+		}
+		if len(*t0s) == 0 {
+			fmt.Println("No T0 found, please create one before")
+			return
+		}
+		if len(*t0s) > 1 && cmd.Flag("t0").Value.String() == "" {
+			fmt.Println("More than one T0 found, please specify one")
+			return
+		}
+		var t0 string
+		if len(*t0s) == 1 {
+			t0 = (*t0s)[0].Tier0Vrf
+		} else {
+			t0, err = cmd.Flags().GetString("t0")
+			if err != nil {
+				fmt.Println("Error to retrieve T0", err)
+				return
+			}
+		}
```

Comment:
Improve this : 

1. is not necessary to GetListT0 if the use set `--t0` flage
2. Simplify the logic